storage/engine/mvcc.go

```diff
@@ -939,14 +947,23 @@ func MVCCIterate(engine Engine, key, endKey proto.Key, timestamp proto.Timestamp
 		return key, iter.ValueProto(msg)
 	}
 
+	// A cumulative write intent error to gather all write intents.
+	var wiErr error
```

Comment:
Actually this is intentional and a byproduct of `var wiErr *proto.WriteIntentError`, when returned as an `error` not being `nil` even if it is nil.